### PR TITLE
Add a :role selector to Capybara, based on data-role attribute

### PR DESCRIPTION
**Pull request by davidstosik**
Alllows:

```html
<li data-role="recipe">
```

```rb

within(:role, "recipe") do
  # ...
end

expect(page).to have_role?("recipe")

expect(page).to have_role?("recipe", count: 20)

page.find_all(:role, "recipe")
```

### CODE REVIEW

File: lib/utensils/custom_matchers.rb [hunk 1]
```diff
@@ -146,6 +146,16 @@ def failure_message_when_negated
...
       Ordered.new(args)
     end
+
+    class HaveRole < Capybara::RSpecMatchers::HaveSelector
+      def initialize(role, *args, &filter_block)
+        super(:role, role, *args, &filter_block)
+      end
+    end
+
+    def have_role(*args)
```
Comment (davidstosik) [inline lib/utensils/custom_matchers.rb]: All other `have_` methods in this file are not taking a question mark, so I reused same convention.

**Comment by davidstosik** [comment]
@balvig Should I close this?

**Comment by balvig** [comment]
@davidstosik thanks a lot for this! I guess based on our conclusions in https://github.com/cookpad/guides/pull/49 we won't need it just yet (could change!), so maybe ok to close for now?

**Comment by davidstosik** [comment]
Okay!

**Pull request closed without merging**